test/ArcxAnalyticsSdk.test.ts

```diff
@@ -324,12 +334,48 @@ describe('(unit) ArcxAnalyticsSdk', () => {
     })
 
     describe('#_trackPagesChange', () => {
-      it('registers an on-click event listener', () => {
-        const addEventListenerStub = sinon.stub(document.body, 'addEventListener')
+      it('registers a locationChange event', () => {
+        const pageStub = sinon.stub(analyticsSdk, 'page')
+        analyticsSdk['_trackPagesChange']()
 
+        window.dispatchEvent(
+          new window.Event('locationchange', { bubbles: true, cancelable: false }),
+        )
+
+        expect(pageStub).calledWithExactly({ url: TEST_JSDOM_URL })
+      })
+
+      it('triggers a locationchange event on history.pushState', () => {
+        const locationChangeListener = sinon.spy()
+        analyticsSdk['_trackPagesChange']()
+
+        window.addEventListener('locationchange', locationChangeListener)
+        window.history.pushState({}, '', '/new-url')
+        expect(locationChangeListener).calledOnce
+
+        window.removeEventListener('locationchange', locationChangeListener)
+      })
+
+      it('triggers a locationchange event on history.replaceState', () => {
+        const locationChangeListener = sinon.spy()
         analyticsSdk['_trackPagesChange']()
 
-        expect(addEventListenerStub).calledOnce
+        window.addEventListener('locationchange', locationChangeListener)
+        window.history.replaceState({}, '', '/new-url')
+        expect(locationChangeListener).calledOnce
+
+        window.removeEventListener('locationchange', locationChangeListener)
+      })
+
+      it('triggers a locationchange event on history.popstate', () => {
+        const locationChangeListener = sinon.spy()
+        analyticsSdk['_trackPagesChange']()
+
+        window.addEventListener('locationchange', locationChangeListener)
```

Comment:
I think better solution is extract locationChange handler inside sdk , spy him, and check if exactly needed handler is called